blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionHandler.java

```diff
@@ -166,7 +157,14 @@ public void handle(ChannelHandlerContext ctx, HttpRequest request) {
                 jsonTimerContext.stop();
             }
 
-            if (containerMetrics == null || containerMetrics.isEmpty()) {
+            // verify metrics after parsing json and converting to metrics
+            List<String> errors = jsonMetricsContainer.getValidationErrors();
+            if ( !errors.isEmpty() && containerMetrics.isEmpty()) {
+                // there was validation errors and all metrics were invalid
+                DefaultHandler.sendResponse( ctx, request, getResponseBody( errors ), HttpResponseStatus.BAD_REQUEST );
+                return;
+            } else if (containerMetrics == null || containerMetrics.isEmpty()) {
+                // empty container
                 log.warn(ctx.getChannel().getRemoteAddress() + " No valid metrics");
                 DefaultHandler.sendResponse(ctx, request, "No valid metrics", HttpResponseStatus.BAD_REQUEST);
```

Comment:
good catch, other if in place prevented bad things from happening after, but safer to add it.